packages/cozy-client/src/CozyClient.js

```diff
@@ -198,42 +198,54 @@ class CozyClient {
       event: TRIGGER_CREATION
     }
   ) {
-    if (!document._type) return document
-
-    const schema = this.schema.getDoctypeSchema(document._type)
-    if (!schema.cozyMetadata) return document
-
-    const enriched = {
-      ...document
-    }
-    const metadata = Object.entries(schema.cozyMetadata)
-
-    for (const [
-      metadataName,
-      { trigger, value, useCurrentDate = false }
-    ] of metadata) {
-      const shouldSetMetadata =
-        options.event === TRIGGER_CREATION ||
-        (options.event === TRIGGER_UPDATE && trigger === TRIGGER_UPDATE)
-      if (!shouldSetMetadata) continue
-
-      let newValue
-      const metadataPath = `cozyMetadata.${metadataName}`
-
-      const shouldAccumulateValues = Array.isArray(value)
-      if (shouldAccumulateValues) {
-        const prevValue = get(document, metadataPath, [])
-        newValue = uniqBy([...prevValue, ...value])
-      } else if (useCurrentDate) {
-        newValue = new Date().toISOString()
-      } else {
-        newValue = get(document, metadataPath, value)
+    if (!document._type || this.metadata === undefined) return document
+
+    const {
+      doctypeVersion,
+      slug,
+      sourceAccount = null,
+      version
+    } = this.metadata
+
+    const now = new Date().toISOString()
+
+    let cozyMetadata = get(document, 'cozyMetadata', {})
+    if (options.event === TRIGGER_CREATION) {
+      cozyMetadata = {
+        ...cozyMetadata, // custom metadata that are set by the app
+        doctypeVersion,
+        createdByApp: slug,
+        sourceAccount,
+        createdAt: now,
+        createdByAppVersion: version,
+        updatedAt: now,
+        updatedByApps: [
+          {
+            date: now,
+            slug
+          }
+        ]
+      }
+    } else if (options.event === TRIGGER_UPDATE) {
+      cozyMetadata = {
+        // add doctypeVersion and sourceAccount if they don't exist
+        doctypeVersion,
+        sourceAccount,
```

Comment:
Not sure we should define a `sourceAccount`. If my connector is updating pre-existing data, we should probably not say that these updated documents come from the connector account.

Reply:
Ok I removed it. Do we want to update it if e.g the contact is attached to a contact account later or is it only if the contact has been created via the konnector? From what I understand here, it's only for creation (and we still have the id of the account in the `io.cozy.contacts.accounts` so it should be OK)